File: 2014/Round_1A/C_Proper_Shuffle/ProperShuffle.py
import sys

def debug(x):
#    print(x)
    return

def solve(t):
    N = int(input())
    data = [int(d) for d in input().split(' ')]

    count = 0
    for n in range(N):
        if int(data[n]) <= n:
            count = count + 1

    print(("Case #%d: " % (t + 1)) + ("BAD" if count < 485 else "GOOD"))

T = int(input())

for t in range(T):
    solve(t)

File: 2014/Round_1A/C_Proper_Shuffle/ProperShuffle.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Based on official analysis
void solve(int t)
{
    int N;
    scanf("%d", &N);

    int count = 0;
    int v;
    for (int n = 0; n < N; n++) {
        scanf("%d", &v);
        if (v <= n) {
            count++;
        }
    }

    printf("Case #%d: %s\n", t + 1, count < 485 ? "BAD" : "GOOD" );
}

int main(int argc, char *argv[])
{
    int T;
    scanf("%d", &T);

    for (int t = 0; t < T; t++) {
        solve(t);
    }

    return 0;
}
